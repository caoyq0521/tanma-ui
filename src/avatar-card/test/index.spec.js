import { mount } from '@vue/test-utils';
import TmAvatarCard from '../avatar-card.vue';

describe('tmAvatarCard', () => {
  it('render AvatarCard', () => {
    const wrapper = mount(TmAvatarCard);
    expect(wrapper).toMatchSnapshot();
  });

  it('title', () => {
    const wrapper = mount(TmAvatarCard);
    const title = '标题信息';
    wrapper.setProps({title});
    expect(wrapper.find('.tm-avatar-card__info .info-title__name').text()).toBe(title);
  })

  it('tip', () => {
    const wrapper = mount(TmAvatarCard);
    wrapper.setProps({tip: "提示信息"});
    expect(wrapper.find('.tm-avatar-card__info .info-title__tip').exists()).toBe(true);
  });

  it('hideDescription', () => {
    const wrapper = mount(TmAvatarCard);
    wrapper.setProps({ hideDescription: false });
    expect(wrapper.find('.tm-avatar-card__info .tm-ellipsis.info-description').exists()).toBe(false);
  });
    
  it('description', () => {
    const wrapper = mount(TmAvatarCard);
    const description = '描述信息';
    wrapper.setProps({ description });
    expect(wrapper.find(wrapper.find('.tm-avatar-card__info .info-description span.tm-ellipsis').text()).toBe(description));
  });
  
  it('hideWechatIcon', () => {
    const wrapper = mount(TmAvatarCard);
    wrapper.setProps({ hideWechatIcon: false});
    expect(wrapper.find('.tm-avatar-card__info .wechat-icon').exists()).toBe(false);
  });

})
  