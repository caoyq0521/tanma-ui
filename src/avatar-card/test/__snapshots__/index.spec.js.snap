// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tmAvatarCard render avatar-card 1`] = `
<div class="tm-avatar-card">
  <div class="tm-avatar tm-avatar__square" style="height: 45px; width: 45px;">
    <div class="tm-flex__center tm-avatar__image"><i class="iconfont icon-touxiang" style="font-size: 21px;"></i>
      <!---->
    </div>
  </div>
  <div class="tm-avatar-card__info">
    <div class="info-title">
      <p class="tm-ellipsis info-title__name">我是标题</p>
      <!---->
    </div>
    <div class="tm-ellipsis info-description"><i class="iconfont wechat-icon icon-wechat-unadded color-gray"></i>待添加
    </div>
  </div>
</div>
`;
